src/background.js

```diff
@@ -342,20 +354,63 @@ const rollout = {
     // Determine to skip additional heuristics (by presence of an enterprise policy)
     if (policyEnableDoH === "no_policy_set") {
       // Resetting skipHeuristicsCheck in case a user had a policy and then removed it!
-      await rollout.setSetting("doh-rollout.skipHeuristicsCheck", false);
+      await rollout.setSetting(DOH_SKIP_HEURISTICS_PREF, false);
     } else {
       // Don't check for prefHasUserValue if policy is set to disable DoH
-      await rollout.setSetting("doh-rollout.skipHeuristicsCheck", true);
+      await rollout.setSetting(DOH_SKIP_HEURISTICS_PREF, true);
     }
     return;
 
   },
 
+  async migrateLocalStoragePrefs() {
+
+    // Migrate updated local storage item names. If this has already been done once, skip the migration
+    const isMigrated = await browser.experiments.preferences.getBoolPref(DOH_BALROG_MIGRATION_PREF, false);
+
+    if (isMigrated) {
+      return;
+    }
+
+    // Check all local items, and migrate them to prefs. If no value set, ignore them.
+    const localStorageItems = [
+      "doneFirstRun",
+      "skipHeuristicsCheck",
+      DOH_ENABLED_PREF,
+      TRR_MODE_PREF,
+      DOH_SELF_ENABLED_PREF,
+      DOH_PREVIOUS_TRR_MODE_PREF,
+      DOH_DOORHANGER_SHOWN_PREF,
+      DOH_DOORHANGER_USER_DECISION_PREF,
+      DOH_DISABLED_PREF,
+      DOH_SKIP_HEURISTICS_PREF,
+      DOH_DONE_FIRST_RUN_PREF
+    ];
+
+    for (let item of localStorageItems) {
+      let data = await browser.storage.local.get(item);
+      let value = data[item];
+
+      let migratedName = name;
+
+      if (!name.startsWith("doh-rollout.")){
+        migratedName = "doh-rollout." + name;
+      }
+
+      if (value !== undefined) {
```

Comment:
nit: We can move this null-check higher up above `let migratedName = name;` - and `continue` if `value === undefined`. And for next-level precision, I suggest checking `!data.hasOwnProperty(item)` instead. 😄

Reply:
Revised. I wrapped all the logic inside the if statement so that the function only applies to items that hold a value (aren't undefined).